src/components/Dialog/Dialog.tsx

```diff
@@ -200,23 +201,24 @@ export default class Dialog extends Component<DialogProps, DialogState> {
     }
   }
 
-  getImages = async (query: string) => {
+  getImages = async (query: string, error: IxError) => {
     const assets = await this.state.imgix.request(
       `assets/${this.state.selectedSource?.id}${query}`,
     );
     // TODO: add more explicit types for image
     this.handleTotalImageCount(
       parseInt((assets.meta.cursor as any).totalRecords || 0),
+      error,
     );
     return assets;
   };
 
-  getImagePaths = async (query: string) => {
+  getImagePaths = async (query: string, error: IxError) => {
```

Comment:
Errors are usually thrown and caught, but it's not clear why we'd need to pass errors as arguments. What's the reasoning here?

Reply:
There's a component in Contentful called a `Note` which we use to surface certain errors in a user-friendly format. Using a variable in state, we track an array of these errors types so we know which one to render on the UI. Here is an example of one:

<img width="1095" alt="Screen Shot 2021-11-12 at 2 45 07 PM" src="https://user-images.githubusercontent.com/15919091/141571917-a8abc509-bf4c-400f-8ba0-07cbec05b85f.png">

Before we set out to implement search, there was only one path that `requestImageUrls()` could have taken: to request the default set of images for a given source (ie no filtering involved). But now that we have search, this function can be used for one of two reasons: the aforementioned default image set or a filtered view of images based on a given search term. Since both of these pathways have their own respective error states, we need to signify which error type should be surfaced if the given operation returns an empty list of images. We do this by passing the error type down when making the request.